src/impurity.jl

```diff
@@ -149,19 +155,25 @@ function impurity_density_model(meta::PropDict, crystal_metadata::PropDict, fit_
 
     T = Float32
     dist = T.(crystal_metadata.impurity_measurements.distance_from_seed_end_mm)
-    L = dist[end] - dist[1] #mm
+    # sometimes might not have measurements at seed end? but last measurement can be taken as crystal length
+    L = dist[end] #mm
 
     # points in crystal axis from 0 to L with step of 1 mm
-    Npoints = Int(ceil(L))
-    zpoints = LinRange(0, L, Npoints)
-	imp = Array{Float32}(undef, Npoints, 1) # impurities in 10^10 cm^-3
+    # always put 200 points to make all .dat files same length
+	imp = Array{Float32}(undef, 200, 1) # impurities in 10^10 cm^-3
 
+    # length is L+1 including both ends
+    Npoints = Int(ceil(L))+1 # step of 1 mm
+    zpoints = LinRange(0, L, Npoints) 
     for i in 1:Npoints
         # divide by 10 to convert from 1e9 e/cm^3 to 1e10 e/cm^3
         # invert the order of the values because with siggen z=0 is tail (dirtiest part)
         # i.e. z = L in our system (Mirion measurements)
         imp[Npoints-i+1] = -(a + b * zpoints[i] + c * exp((zpoints[i] - L)/tau)) / 10.
     end
+    for i in Npoints+1 : 200
+        imp[i] = 0
+    end
```

Comment:
Here, you could just define 
```julia
imp = zeros(Float32, 200)
```
and then only run the first loop
```julia
    for i in 1:Npoints
        # divide by 10 to convert from 1e9 e/cm^3 to 1e10 e/cm^3
        # divide by 10 to convert from 1e9 e/cm^3 to 1e10 e/cm^3
        # invert the order of the values because with siggen z=0 is tail (dirtiest part)
        # invert the order of the values because with siggen z=0 is tail (dirtiest part)
        # i.e. z = L in our system (Mirion measurements)
        # i.e. z = L in our system (Mirion measurements)
        imp[Npoints-i+1] = -(a + b * zpoints[i] + c * exp((zpoints[i] - L)/tau)) / 10.
        imp[Npoints-i+1] = -(a + b * zpoints[i] + c * exp((zpoints[i] - L)/tau)) / 10.
    end
```
Then, there would be no need for an additional loop
```julia
    for i in Npoints+1 : 200
        imp[i] = 0
    end
```

Reply:
Oh thanks that makes much more sense! Changed it to be that way.